registration/round.go

```diff
@@ -180,14 +181,18 @@ func (r *round) submit(ctx context.Context, key, challenge []byte) (<-chan error
 	}
 
 	if r.batch.Len() >= r.maxBatchSize {
-		r.flushPendingSubmitsLocked()
+		r.flushPendingSubmitsLocked(false)
 	} else if r.batch.Len() == 1 {
 		logging.FromContext(ctx).Debug(
 			"scheduling flush of pending submits",
 			zap.Uint("round", r.epoch),
 			zap.Duration("interval", r.flushInterval),
 		)
-		time.AfterFunc(r.flushInterval, r.flushPendingSubmits)
+		r.pendingFlush = time.AfterFunc(r.flushInterval, func() {
+			r.batchMutex.Lock()
+			defer r.batchMutex.Unlock()
+			r.flushPendingSubmitsLocked(true)
```

Comment:
The boolean parameter isn't really needed or is it?

```suggestion
			r.batchMutex.Lock()
			defer r.batchMutex.Unlock()
			if r.pendingFlush != nil {
				r.flushPendingSubmitsLocked()
			}
```

You can even put this into its own method to make calling it from the test easier 🙂 

```go
func (r *round) timedFlushPendingSubmits() {
	r.batchMutex.Lock()
	defer r.batchMutex.Unlock()
	if r.pendingFlush != nil {
		r.flushPendingSubmitsLocked()
	}
}
```